### PR TITLE
Transaction

### PR DESCRIPTION
**Pull request by thyerre**

### CODE REVIEW

File: app/Account.php [hunk 1]
```diff
@@ -3,10 +3,53 @@
...
 use Illuminate\Database\Eloquent\Model;
+use \App\Transaction;
 
 class Account extends Model
 {
     protected $table = 'account';
     protected $primaryKey = 'id';
     protected $fillable = ['id', 'description', 'id_banking', 'current_balance', 'id_user'];
+
+    public static function alterBalance($request)
```
Comment (DevKleber) [inline app/Account.php:14]:
```suggestion
    public static function updateBalance($request)
```

File: app/Account.php [hunk 2]
```diff
@@ -3,10 +3,53 @@
...
+        if ($request['is_income']) {
+            $ar['current_balance'] = $account->current_balance + $request['value'];
+        } else {
+            $ar['current_balance'] = $account->current_balance - $request['value'];
+        }
+
+        return $account->update($ar);
+    }
+
+    public static function returnBalanceByTransaction(int $id, Transaction $transaction)
```
Comment (DevKleber) [inline app/Account.php:35]:
```suggestion
    public static function getBalanceBackByTransaction(int $id, Transaction $transaction)
```

File: app/Http/Controllers/TransactionController.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        }
+
+        return response($transaction);
+    }
+
+    public function store(Request $request)
+    {
+        $ar = $request->all();
+        $ar['id_user'] = auth('api')->user()->id;
+
```
Comment (DevKleber) [inline app/Http/Controllers/TransactionController.php:25]: precisa iniciar uma transação. 
\DB::beginTransaction();
\DB::rollBack();
\DB::commit();

**Review by DevKleber — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/TransactionController.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    {
+        \DB::beginTransaction();
+
+        $ar = $request->all();
+        $ar['id_user'] = auth('api')->user()->id;
+
+        $transaction = \App\Transaction::create($ar);
+
+        if (!$transaction) {
+            return response(['message' => 'Erro ao salvar Transação'], 400);
```
Comment (DevKleber) [inline app/Http/Controllers/TransactionController.php:31]: \DB::rollBack();

File: app/Http/Controllers/TransactionController.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        $arTransactionAccount['account_id'] = $ar['account_id'];
+        $arTransactionAccount['transaction_id'] = $transaction->id;
+
+        $transactionAccount = \App\TransactionAccount::create($arTransactionAccount);
+
+        if (!$transactionAccount) {
+            return response(['message' => 'Erro ao salvar transaction account'], 400);
```
Comment (DevKleber) [inline app/Http/Controllers/TransactionController.php:40]: \DB::rollBack();

File: app/Http/Controllers/TransactionController.php [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        if (!\App\TransactionItem::saveItens($ar, $transaction)) {
+            return response(['message' => 'Erro ao salvar itens'], 400);
```
Comment (DevKleber) [inline app/Http/Controllers/TransactionController.php:44]: \DB::rollBack();

File: app/Http/Controllers/TransactionController.php [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        if (!\App\Account::updateBalance($ar)) {
+            return response(['message' => 'Erro ao alterar valor da conta'], 400);
```
Comment (DevKleber) [inline app/Http/Controllers/TransactionController.php:48]: \DB::rollBack();

**Review by DevKleber — CHANGES_REQUESTED** [review]
\DB::rollBack();

**Comment by thyerre** [comment]
[Income Transaction](https://trello.com/c/ajSV6Dc3/8-income-transaction)

**Review by DevKleber — APPROVED** [review]

**Pull request merged**